video2dataset/data_reader.py

```diff
@@ -3,9 +3,88 @@
 import uuid
 import requests
 import yt_dlp
+import io
 
+try:
+    import webvtt  # pip install webvtt-py
+except ImportError as err:
+    print(err)
 
-def handle_youtube(youtube_url, tmp_dir, video_height, video_width):
+
+def sub_to_dict(sub, dedupe=True, single=False) -> list:
+    """Convert WebVTT to JSON, optionally removing duplicate lines"""
+
+    captions = webvtt.read_buffer(io.StringIO(sub))
+    dicts = [{"start": c.start, "end": c.end, "lines": c.lines} for c in captions]
+    if dedupe:
+        dicts = []
+        prev_line = None
+        for c in captions:
+            if any("<c>" in l for l in c.lines):
+                continue
+            # Collect lines that are not dupes
+            not_dupe_lines = []
+            for line in c.lines:
+                if not line.strip():
+                    continue
+                if line != prev_line:
+                    not_dupe_lines.append(line)
+                prev_line = line
+            if not_dupe_lines:
+                dicts.append({"start": c.start, "end": c.end, "lines": not_dupe_lines})
+    if single:
+        for d in dicts:
+            d["line"] = "\n".join(d.pop("lines"))
+    return dicts
+
+
+def get_yt_meta(url, yt_metadata_args: dict) -> tuple:
+    """Return info dict and/or downloads subtitles
+    yt_metadata_args is a dict of follwing format:
+    yt_metadata_args = {
+        'writesubtitles': True,
+        'subtitleslangs': ['en'],
+        'writeautomaticsub': True,
+        'get_info': True
+    }
+
+    writesubtitles:    Whether to write subtitles
+    writeautomaticsub: Write the automatically generated subtitles to a file
+    subtitleslangs:    List of languages of the subtitles to download.
+    get_info: whether to add info (title, description, tags etc) to the output.
+
+    """
+
+    write_subs = yt_metadata_args.get("writesubtitles", None)
+
+    yt_metadata_args["skip_download"] = True
+    yt_metadata_args["ignoreerrors"] = True
+    yt_metadata_args["quiet"] = True
+
+    info_dict, sub_dict = None, None
+
+    with yt_dlp.YoutubeDL(yt_metadata_args) as yt:
+
+        info_dict = yt.extract_info(url, download=False)
+        if write_subs:
+            sub_url = info_dict["requested_subtitles"][yt_metadata_args["subtitleslangs"][0]]["url"]
+            res = requests.get(sub_url)
+            sub = io.TextIOWrapper(io.BytesIO(res.content)).read()
+            sub_dict = sub_to_dict(sub)
+
+        if yt_metadata_args["get_info"]:
+            info_dict.pop("subtitles")
+            info_dict.pop("requested_formats")
+            info_dict.pop("formats")
+            info_dict.pop("thumbnails")
+            info_dict.pop("automatic_captions")
+        else:
+            info_dict = None
+
+        return info_dict, sub_dict
```

Comment:
instead of returning sub_dict as a separate output why don't we just make it an entry in info_dict. I don't think subtitles need a dedicated output variable

Reply:
Yeah, I also initially thought of putting subtitles into info_dict, but there're might some problems:
1. You can specify get_info=False so returning info_dict with subtitles in such case would be misleading imo
2. I plan to implement the feature that will allow video to be split into segments with a segment for each subtitle sentence so I'm going to use sub_dict for this and it would be not really good to pass the whole info_dict every time.

Reply:
1. right so we have data_reader to not output info_dict and sub_dict but instead a yt_meta_dict which has information from both
2. then you could extract your sub_dict from the yt_meta_dict

this is very minimal, I just don't want us to get into the habit of having functions with many outputs where the category of all the output are the same - YouTube metadata about the video (both the subtitles and other info fit this definition)

Reply:
Ok